teos/src/config.rs

```diff
@@ -203,6 +203,12 @@ impl Config {
         self.tor_support |= options.tor_support;
         self.debug |= options.debug;
         self.deps_debug |= options.deps_debug;
+        if self.overwrite_key {
+            panic!("The 'overwrite_key' parameter cannot be set in the configuration file. Please pass it as a command-line argument if necessary.");
```

Comment:
Don't panic. Make the method return a `Result` and check it on main, if it's an error,  `eprintln!("{e}")` and `std::process::exit(1)`

Reply:
> Don't panic. Make the method return a `Result` and check it on main, if it's an error, `eprintln!("{e}")` and `std::process::exit(1)`

if path_with_options have to updated with result, I have to add `Result<(), String>` in the function to make the error message work. But the problem is I have to change all the code path function to return a result.

Reply:
> Don't panic. Make the method return a `Result` and check it on main, if it's an error, `eprintln!("{e}")` and `std::process::exit(1)`
<img width="1080" alt="Screenshot 2023-05-12 at 10 50 54 AM" src="https://github.com/talaia-labs/rust-teos/assets/74004409/65ed573b-37c4-4a31-a955-2662045a5888">

Reply:
You should do such sanity checks in `config::from_file` instead of `patch_with_options`.